x-pack/plugins/security_solution/public/management/pages/blocklist/view/components/create_blocklist_form.tsx

```diff
@@ -0,0 +1,472 @@
+        onChange({
+          isValid: isValid && !nameErrors.length,
+          item: {
+            ...item,
+            name,
+          },
+        });
+      },
+      [errors, onChange, isValid, item]
+    );
+
+    const handleOnDescriptionChange = useCallback(
+      (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+        onChange({
+          isValid,
+          item: {
+            ...item,
+            description: event.target.value,
+          },
+        });
+      },
+      [onChange, isValid, item]
+    );
+
+    const handleOnOsChange = useCallback(
+      (os: OperatingSystem) => {
+        const nextIsValid = validate(
+          blocklistEntry.field,
+          os,
+          blocklistEntry.type,
+          blocklistEntry.value
+        );
+
+        onChange({
+          isValid: nextIsValid,
+          item: {
+            ...item,
+            os_types: [os],
+            entries: [
+              {
+                ...blocklistEntry,
+                field:
+                  os !== OperatingSystem.WINDOWS &&
+                  blocklistEntry.field === ConditionEntryField.SIGNER
+                    ? ConditionEntryField.HASH
+                    : blocklistEntry.field,
+              },
+            ],
+          },
+        });
+      },
+      [validate, blocklistEntry, onChange, item]
+    );
+
+    const handleOnFieldChange = useCallback(
+      (field: ConditionEntryField) => {
+        const nextIsValid = validate(field, selectedOs, blocklistEntry.type, blocklistEntry.value);
+
+        onChange({
+          isValid: nextIsValid,
+          item: {
+            ...item,
+            entries: [{ ...blocklistEntry, field }],
+          },
+        });
+      },
+      [validate, onChange, item, selectedOs, blocklistEntry]
+    );
+
+    // only triggered on remove / clear
+    const handleOnValueChange = useCallback(
+      (change: Array<EuiComboBoxOptionOption<string>>) => {
+        const value = change.map((option) => option.label);
+
+        const nextIsValid = validate(blocklistEntry.field, selectedOs, blocklistEntry.type, value);
+
+        onChange({
+          isValid: nextIsValid,
+          item: {
+            ...item,
+            entries: [{ ...blocklistEntry, value }],
+          },
+        });
+      },
+      [validate, onChange, item, selectedOs, blocklistEntry]
+    );
+
+    const handleOnValueAdd = useCallback(
+      (option: string) => {
+        const splitValues = option.split(',').filter((value) => value.trim());
+        const value = [...blocklistEntry.value, ...splitValues];
+
+        const nextIsValid = validate(blocklistEntry.field, selectedOs, blocklistEntry.type, value);
+
+        onChange({
+          isValid: nextIsValid,
+          item: {
+            ...item,
+            entries: [{ ...blocklistEntry, value }],
+          },
+        });
+      },
+      [validate, onChange, item, selectedOs, blocklistEntry]
+    );
+
+    const handleOnPolicyChange = useCallback(
```

Comment:
Not sure if you already fixed the problem with the policy selector, if not, [check this code](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/view/components/form.tsx#L120-L137), it should help you